### DISCUSSION TITLE
Possible Bug?

### DISCUSSION THREAD

**Discussion by Oliver-Turp** [prisma/prisma#21585] [Q&A]
### Question

Can you tell me if this is a bug? I have a many to many connection between my User model and my discord Roles model. When I delete a User from the database, they are [still visible](https://github.com/prisma/prisma/assets/55669423/c63d54c2-59e1-460e-bdcc-dd5122df6eab) on the prisma studio localhost web page as connected each Role they had... However, if I console log the user Role connections, the connections from delete users are [correctly gone](https://github.com/prisma/prisma/assets/55669423/28b55c8e-1ec0-4a1f-ab90-d8b27784b340). I assume this is a cache thing. However, I have restarted the webpage and restart the npx prisma studio connection... Please advise

### Expected behavior (optional)

I'd expect the console logged data to be replicated with what I find in the studio.

### Information about Prisma Schema, Client Queries and Environment (optional)
```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider     = "mysql"
  url          = env("DATABASE_URL")
  relationMode = "prisma"
}

model Rules {
  ruleId         String    @id @unique @map("_ruleId")
  ruleIdentifier String    @unique @db.VarChar(255)
  ruleTitle      String    @db.VarChar(255)
  ruleDesc       String    @db.VarChar(500)
  ruleReason     String    @db.VarChar(255)
  ruleNum        Int       @unique
  ruleType       String    @db.VarChar(255)
  createdAt      DateTime  @default(now())
  updatedAt      DateTime  @updatedAt
  warnings       Warning[]
}

model User {
  userId          BigInt          @id @unique
  username        String          @db.VarChar(500)
  avatar          String?
  status          String?         @default("Active")
  createdAt       DateTime        @default(now())
  updatedAt       DateTime        @updatedAt
  warnUser        Warning[]       @relation("Target")
  warnExe         Warning[]       @relation("Executor")
  warnReviewer    Warning[]       @relation("Reviewer")
  leaderboard     Leaderboard[]
  cooldowns       UserCooldowns[]
  XPLogsRecipient XPLogs[]        @relation("Recipient")
  XPLogsProvider  XPLogs[]        @relation("Provider")
  roles           Roles[]         @relation("UserRoles")
}

model Roles {
  roleId    BigInt   @id @unique
  roleName  String
  visible Boolean
  color String
  position Int?
  users     User[]   @relation("UserRoles")
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}

model Warning {
  warnId         Int      @id @unique @default(autoincrement())
  reason         String   @db.VarChar(500)
  executor       User     @relation("Executor", fields: [executorId], references: [userId], onDelete: Cascade)
  executorId     BigInt
  target         User     @relation("Target", fields: [targetId], references: [userId], onDelete: Cascade)
  targetId       BigInt
  ruleIdentifier String   @db.VarChar(255)
  rule           Rules    @relation(fields: [ruleIdentifier], references: [ruleIdentifier])
  needsReview    Boolean
  underReview    Boolean
  reviewer       User?    @relation("Reviewer", fields: [reviewerId], references: [userId])
  reviewerId     BigInt?
  embedURL String?
  createdAt      DateTime @default(now())
  updatedAt      DateTime @updatedAt

  @@index([targetId])
  @@index([ruleIdentifier])
  @@index([executorId])
  @@index([reviewerId])
}

model Leaderboard {
  position  Int?
  user      User     @relation(fields: [userId], references: [userId], onDelete: Cascade)
  userId    BigInt   @unique
  level     Int      @default(0)
  xp        Int      @default(0)
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  @@index([userId])
}

model XPLogs {
  id          Int      @id @default(autoincrement())
  recipient   User     @relation("Recipient", fields: [recipientId], references: [userId], onDelete: Cascade)
  recipientId BigInt
  provider    User     @relation("Provider", fields: [providerId], references: [userId], onDelete: Cascade)
  providerId  BigInt
  xpChange    Int
  timestamp   DateTime @default(now())

  @@index([recipientId])
  @@index([providerId])
}

model UserCooldowns {
  id              Int       @id @default(autoincrement())
  user            User      @relation(fields: [userId], references: [userId], onDelete: Cascade)
  userId          BigInt    @unique
  msgXP           DateTime?
  thanksCount     Int       @default(0)
  lastThank       DateTime?
  thanksLockedOut DateTime?

  @@index([userId])
}

model ChannelCooldowns {
  id            Int       @id @default(autoincrement())
  channelId     String    @unique
  stickyMessage DateTime?
  messageCount  Int       @default(0)
}
```

- OS: Windows
- Database: MySQL & PlanetScale
- Node.js version: v18.17.1
- Prisma version: v5.4.2

**Comment by ludralph** [comment]
Hi @Oliver-Turp 👋 

Thank you for raising this question.

Can you please open Prisma Studio in Incognito mode on your browser to see if you get the same old data? You may also try to clear the cache in your browser to see if you get the same stale data.

**Reply by Oliver-Turp** [reply]
Hi,

Thanks for getting back to me @ludralph 

I tried clearing all the browing data, and tried on 3 different browsers (Chrome, Firefox and edge) both normal and incognito. Only difference now is the User data attached to the roles [is all empty](https://github.com/prisma/prisma/assets/55669423/5707eafe-6cae-4e79-a785-83bd591e1229)

Please advise.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
